src/Compilers/Shared/NamedPipeUtil.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Security.AccessControl;
+using System.Security.Principal;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// The compiler needs to take advantage of features on named pipes which require target framework
+    /// specific APIs. This class is meant to provide a simple, universal interface on top of the 
+    /// multi-targeting code that is needed here.
+    /// </summary>
+    internal static class NamedPipeUtil
+    {
+        const int s_currentUserOnlyValue = unchecked((int)0x20000000);
```

Comment:
consider scoping to `#if NET472` block

Reply:
Will do. Have a follow up PR coming and I will put this into that.